src/metabase/models/persisted_info.clj

```diff
@@ -135,20 +135,21 @@
 
                          (contains? #{"deletable" "off"} (:state existing-persisted-info))
                          (do
-                           (db/update! PersistedInfo (u/the-id existing-persisted-info)
-                                       :active false, :state "creating", :state_change_at :%now)
+                           (t2/update! PersistedInfo (u/the-id existing-persisted-info)
+                                       {:active false, :state "creating", :state_change_at :%now})
                            (t2/select-one PersistedInfo :card_id card-id)))]
     persisted-info))
 
 (defn ready-database!
   "Sets PersistedInfo state to `creating` for models without a PeristedInfo or those in a `deletable` state.
    Will ignore explicitly set `off` models."
   [database-id]
-  (db/update! PersistedInfo
-              {:where [:and
-                       [:= :database_id database-id]
-                       [:= :state "deletable"]]
-               :set {:active false,
-                     :state "creating",
-                     :state_change_at :%now}})
+  (t2/query-one
```

Comment:
we're sure it's just one?

Reply:
yep. checked twice on the diff and it's the only one with this call type.